### PR TITLE
feat: Implement basic submit photo screen

### PR DESCRIPTION
**Pull request by lrupnikk**

### CODE REVIEW

File: brd-android/kyc/src/main/java/com/fabriik/kyc/ui/features/submitphoto/SubmitPhotoContract.kt [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import com.fabriik.kyc.data.enums.DocumentType
+
+interface SubmitPhotoContract {
+
+    sealed class Event : FabriikContract.Event {
+        object BackClicked : Event()
+        object RetakeClicked : Event()
+        object ConfirmClicked : Event()
+
+        data class OnCreate(
```
Comment (roknovak) [inline brd-android/kyc/src/main/java/com/fabriik/kyc/ui/features/submitphoto/SubmitPhotoContract.kt]: Event not needed

File: brd-android/kyc/src/main/java/com/fabriik/kyc/ui/features/submitphoto/SubmitPhotoFragment.kt [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+class SubmitPhotoFragment : Fragment(),
+    FabriikView<SubmitPhotoContract.State, SubmitPhotoContract.Effect> {
+
+    private lateinit var binding: FragmentSubmitPhotoBinding
+    private val viewModel: SubmitPhotoViewModel by viewModels()
+    private val args by navArgs<SubmitPhotoFragmentArgs>()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        viewModel.setEvent(SubmitPhotoContract.Event.OnCreate(args.documentType, args.imageUri))
```
Comment (roknovak) [inline brd-android/kyc/src/main/java/com/fabriik/kyc/ui/features/submitphoto/SubmitPhotoFragment.kt]: Not needed

File: brd-android/kyc/src/main/java/com/fabriik/kyc/ui/features/submitphoto/SubmitPhotoFragment.kt [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+import com.fabriik.kyc.data.enums.DocumentType
+import com.fabriik.kyc.databinding.FragmentSubmitPhotoBinding
+import kotlinx.coroutines.flow.collect
+
+class SubmitPhotoFragment : Fragment(),
+    FabriikView<SubmitPhotoContract.State, SubmitPhotoContract.Effect> {
+
+    private lateinit var binding: FragmentSubmitPhotoBinding
+    private val viewModel: SubmitPhotoViewModel by viewModels()
+    private val args by navArgs<SubmitPhotoFragmentArgs>()
```
Comment (roknovak) [inline brd-android/kyc/src/main/java/com/fabriik/kyc/ui/features/submitphoto/SubmitPhotoFragment.kt]: Not needed

File: brd-android/kyc/src/main/java/com/fabriik/kyc/ui/features/submitphoto/SubmitPhotoViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,27 @@
+package com.fabriik.kyc.ui.features.submitphoto
+
+import android.app.Application
+import com.fabriik.common.ui.base.FabriikViewModel
+
+class SubmitPhotoViewModel(
+    application: Application
+
```
Comment (roknovak) [inline brd-android/kyc/src/main/java/com/fabriik/kyc/ui/features/submitphoto/SubmitPhotoViewModel.kt]: Add: `savedStateHandle: SavedStateHandle`

File: brd-android/kyc/src/main/java/com/fabriik/kyc/ui/features/submitphoto/SubmitPhotoViewModel.kt [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+) : FabriikViewModel<SubmitPhotoContract.State, SubmitPhotoContract.Event, SubmitPhotoContract.Effect>(
+    application
```
Comment (roknovak) [inline brd-android/kyc/src/main/java/com/fabriik/kyc/ui/features/submitphoto/SubmitPhotoViewModel.kt]: Pass `savedStateHandle` to superclass

File: brd-android/kyc/src/main/java/com/fabriik/kyc/ui/features/submitphoto/SubmitPhotoViewModel.kt [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+) {
+
```
Comment (roknovak) [inline brd-android/kyc/src/main/java/com/fabriik/kyc/ui/features/submitphoto/SubmitPhotoViewModel.kt:22]: override `parseArguments` method like this:

override fun parseArguments(savedStateHandle: SavedStateHandle) {
        arguments = SubmitPhotoFragmentArgs.fromBundle(
            savedStateHandle.toBundle()
        )
    }

File: brd-android/kyc/src/main/java/com/fabriik/kyc/ui/features/submitphoto/SubmitPhotoViewModel.kt [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+    override fun createInitialState() = SubmitPhotoContract.State()
```
Comment (roknovak) [inline brd-android/kyc/src/main/java/com/fabriik/kyc/ui/features/submitphoto/SubmitPhotoViewModel.kt]: Set data from arguments to State object

File: brd-android/kyc/src/main/java/com/fabriik/kyc/ui/features/submitphoto/SubmitPhotoContract.kt [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+    }
+
+    sealed class Effect : FabriikContract.Effect {
+        object GoBack : Effect()
+        object GoToCamera : Effect()
+        object GoForward : Effect()
+    }
+
+    data class State(
+        val documentType: DocumentType? = null,
```
Comment (roknovak) [inline brd-android/kyc/src/main/java/com/fabriik/kyc/ui/features/submitphoto/SubmitPhotoContract.kt]: Make `documentType` non nullable

File: brd-android/kyc/src/main/java/com/fabriik/kyc/ui/features/submitphoto/SubmitPhotoContract.kt [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+        val image: Uri? = null,
```
Comment (roknovak) [inline brd-android/kyc/src/main/java/com/fabriik/kyc/ui/features/submitphoto/SubmitPhotoContract.kt]: Make `image` non nullable

File: brd-android/kyc/src/main/java/com/fabriik/kyc/ui/features/submitphoto/SubmitPhotoContract.kt [hunk 10]
```diff
@@ -0,0 +1,30 @@
+package com.fabriik.kyc.ui.features.submitphoto
+
+import android.net.Uri
+import com.fabriik.common.ui.base.FabriikContract
+import com.fabriik.kyc.data.enums.DocumentType
+
+interface SubmitPhotoContract {
+
+    sealed class Event : FabriikContract.Event {
+        object BackClicked : Event()
```
Comment (roknovak) [inline brd-android/kyc/src/main/java/com/fabriik/kyc/ui/features/submitphoto/SubmitPhotoContract.kt:11]: Missing DismissClicked event

File: brd-android/kyc/src/main/java/com/fabriik/kyc/ui/features/submitphoto/SubmitPhotoContract.kt [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+        object ConfirmClicked : Event()
+
+        data class OnCreate(
+            val documentType: DocumentType,
+            val image: Uri,
+        ) : Event()
+    }
+
+    sealed class Effect : FabriikContract.Effect {
+        object GoBack : Effect()
```
Comment (roknovak) [inline brd-android/kyc/src/main/java/com/fabriik/kyc/ui/features/submitphoto/SubmitPhotoContract.kt]: Rename to Back

File: brd-android/kyc/src/main/java/com/fabriik/kyc/ui/features/submitphoto/SubmitPhotoContract.kt [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+        object GoToCamera : Effect()
+        object GoForward : Effect()
```
Comment (roknovak) [inline brd-android/kyc/src/main/java/com/fabriik/kyc/ui/features/submitphoto/SubmitPhotoContract.kt]: Rename to PostValidation

File: brd-android/kyc/src/main/java/com/fabriik/kyc/ui/features/submitphoto/SubmitPhotoContract.kt [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+
+        data class OnCreate(
+            val documentType: DocumentType,
+            val image: Uri,
+        ) : Event()
+    }
+
+    sealed class Effect : FabriikContract.Effect {
+        object GoBack : Effect()
+        object GoToCamera : Effect()
```
Comment (roknovak) [inline brd-android/kyc/src/main/java/com/fabriik/kyc/ui/features/submitphoto/SubmitPhotoContract.kt]: Rename to TakePhoto

File: brd-android/kyc/src/main/res/navigation/nav_graph_kyc.xml [hunk 14]
```diff
@@ -66,7 +66,30 @@
...
+            <argument
+                android:name="documentType"
+                app:argType="com.fabriik.kyc.data.enums.DocumentType" />
+
+            <argument
+                android:name="imageUri"
+                app:argType="android.net.Uri" />
+
+            <action
+                android:id="@+id/action_submitPhoto_to_camera"
```
Comment (roknovak) [inline brd-android/kyc/src/main/res/navigation/nav_graph_kyc.xml]: Rename to `actionTakePhoto`

File: brd-android/kyc/src/main/res/navigation/nav_graph_kyc.xml [hunk 15]
```diff
@@ -66,7 +66,30 @@
...
+                app:destination="@id/fragmentPostValidation" />
+
+            <action
+                android:id="@+id/action_submitPhoto_to_postValidation"
```
Comment (roknovak) [inline brd-android/kyc/src/main/res/navigation/nav_graph_kyc.xml]: Rename to `actionPostValidation`

**Review by roknovak — APPROVED** [review]

**Review by dabitdev — APPROVED** [review]
LGTM 👍🏻

**Pull request merged**